s3torchbenchmarking/src/s3torchbenchmarking/benchmark.py

```diff
@@ -71,6 +74,39 @@ def from_conf(dataset_config: DictConfig):
             return DatasetSharding[dataset_config.sharding]
 
 
+def parse_s3_url(url: str) -> (str, str):
+    """Parse an s3://bucket/key/ URL into a (bucket, key) pair"""
+    if not url.startswith("s3://"):
+        raise Exception(f"URL must start with 's3://': {url}")
+    bucket, prefix = url[len("s3://") :].split("/", maxsplit=1)
+    if prefix and not prefix.endswith("/"):
+        raise Exception(f"non-empty prefix must end with '/': {url}")
+    if not bucket:
+        raise Exception(f"invalid bucket name: {url}")
+    return bucket, prefix
+
+
+def make_mountpoint(
+    prefix_uri: str,
+    mountpoint_path: Optional[str] = None,
+    additional_args: Optional[str] = None,
+) -> str:
+    def teardown(path: str):
+        subprocess.run(["sudo", "umount", path])
+        shutil.rmtree(path)
+
+    bucket, prefix = parse_s3_url(prefix_uri)
+    # Run Mountpoint in background mode, and arrange for it to unmount when this script exits
+    tempdir = tempfile.mkdtemp(prefix="s3dataset_")
+    binary = mountpoint_path or "mount-s3"
+    args = additional_args.split(" ") if additional_args else []
+    subprocess.run([binary, bucket, tempdir] + args, check=True)
+    atexit.register(teardown, tempdir)
```

Comment:
Do we want to do this before starting MP?

Reply:
Eager teardown registration sounds reasonable if we're anticipating some surface area of error between the mounting and registration - in this case they are executed one after the other - if the subprocess cmd failed, we shouldn't need to teardown anything.